cpp/oneapi/dal/algo/triangle_counting/common.hpp

```diff
@@ -0,0 +1,195 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/graph/undirected_adjacency_vector_graph.hpp"
+#include "oneapi/dal/table/common.hpp"
+
+namespace oneapi::dal::preview {
+namespace triangle_counting {
+
+namespace task {
+namespace v1 {
+struct local {};
+struct global {};
+struct local_and_global {};
+using by_default = local;
+} // namespace v1
+using v1::local;
+using v1::global;
+using v1::local_and_global;
+using v1::by_default;
+} // namespace task
+
+namespace method {
+namespace v1 {
+struct ordered_count {};
+using by_default = ordered_count;
+} // namespace v1
+using v1::ordered_count;
+using v1::by_default;
+} // namespace method
+
+namespace v1 {
+// Kind of triangles to compute
+enum class kind { undirected_clique, directed_cycle, directed_closed_triplet };
+
+// Option to allow relabeling that is potentially additional memory consuming.
+enum class relabel { yes, no };
+} // namespace v1
+
+using v1::kind;
+using v1::relabel;
+
+namespace detail {
```

Comment:
to double-check - this contains implementation details. Do we need 'detail' folder somewhere?

Reply:
Looks all algorithms have the same structure, aligned